src/Features/Core/Portable/AddDebuggerDisplay/AbstractAddDebuggerDisplayCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.AddImports;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.AddDebuggerDisplay
+{
+    internal abstract class AbstractAddDebuggerDisplayCodeRefactoringProvider<TTypeDeclarationSyntax, TMethodDeclarationSyntax> : CodeRefactoringProvider
```

Comment:
💡  This does not need to be abstract. Shared code exists for adding an attribute to a type declaration. See [TestExportsShouldNotBeDiscoverableCodeFixProvider.cs](https://github.com/dotnet/roslyn-analyzers/blob/16eef22201766b6802d32d338d65f34fa070e008/src/Roslyn.Diagnostics.Analyzers/Core/TestExportsShouldNotBeDiscoverableCodeFixProvider.cs) for an example, which adds `[PartNotDiscoverable]` to types.

Reply:
There are already no abstract methods, but I didn't realize that it was possible to seal this one and register it for both languages.

Reply:
@sharwell This class is currently abstract because the type parameters are specialized for each language. Is there another way I could implement the same behavior?

Reply:
I think this is a moot point now that I have to introduce an abstract member so that `NameOf` is not generated for VB since NameOf is not legal to use on private members from attribute arguments. (If I'm wrong and there's shared code that can do these things, please let me know!)